internal/resources/tanzupackagerepository/data_source_repository.go

```diff
@@ -0,0 +1,85 @@
+/*
+Copyright © 2023 VMware, Inc. All Rights Reserved.
+SPDX-License-Identifier: MPL-2.0
+*/
+
+package tanzupackagerepository
+
+import (
+	"context"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/pkg/errors"
+
+	"github.com/vmware/terraform-provider-tanzu-mission-control/internal/authctx"
+	clienterrors "github.com/vmware/terraform-provider-tanzu-mission-control/internal/client/errors"
+	"github.com/vmware/terraform-provider-tanzu-mission-control/internal/resources/common"
+	"github.com/vmware/terraform-provider-tanzu-mission-control/internal/resources/tanzupackagerepository/scope"
+	"github.com/vmware/terraform-provider-tanzu-mission-control/internal/resources/tanzupackagerepository/spec"
+	"github.com/vmware/terraform-provider-tanzu-mission-control/internal/resources/tanzupackagerepository/status"
+)
+
+func DataSourcePackageRepository() *schema.Resource {
+	return &schema.Resource{
+		Schema: getDataSourceSchema(),
+		ReadContext: func(ctx context.Context, d *schema.ResourceData, m interface{}) diag.Diagnostics {
+			return dataPackageRepositoryRead(context.WithValue(ctx, contextMethodKey{}, DataSourceRead), d, m)
+		},
+	}
+}
+
+func dataPackageRepositoryRead(ctx context.Context, d *schema.ResourceData, m interface{}) (diags diag.Diagnostics) {
+	config := m.(authctx.TanzuContext)
+
+	packageRepositoryName, ok := d.Get(nameKey).(string)
+	if !ok {
+		return diag.Errorf("Unable to read package repository name")
+	}
+
+	packageRepositoryNamespace, ok := d.Get(NamespaceKey).(string)
+	if !ok {
+		return diag.Errorf("Unable to read package repository name")
+	}
+
+	scopedFullnameData := scope.ConstructScope(d, packageRepositoryName, packageRepositoryNamespace)
+
+	if scopedFullnameData == nil {
+		return diag.Errorf("Unable to create Tanzu Mission Control package repository entry; Scope full name is empty")
+	}
+
+	UID, meta, repoSpec, clusterScopeStatus, err := retrievePackageRepositoryUIDMetaAndSpecFromServer(config, scopedFullnameData, d)
+	if err != nil {
+		if clienterrors.IsNotFoundError(err) {
+			if ctx.Value(contextMethodKey{}) == DataSourceRead {
+				return diag.FromErr(errors.Wrapf(err, "Tanzu Mission Control package repository entry not found, name : %s", packageRepositoryName))
+			}
+
+			_ = schema.RemoveFromState(d, m)
+
+			return diags
+		}
+
+		return diag.FromErr(errors.Wrapf(err, "Unable to get Tanzu Mission Control package repository entry, name : %s", packageRepositoryName))
+	}
+
+	if err := d.Set(disabledKey, clusterScopeStatus.Disabled); err != nil {
+		return diag.FromErr(err)
+	}
+
+	d.SetId(UID)
```

Comment:
Do we need this here? It will be set as part of create right?

Reply:
but if we need to get pkg repo but if we are not creating it from terraform then we gonna required this.

Reply:
Sorry I couldn't get this, do you mean the pkg repo that is not created via terraform?

Reply:
yes

Reply:
In order to get such resources into terraform state you have to use terraform import and not like this in my opinion

Reply:
this common practice I've seen throughout the multiple resources